### PR TITLE
Notifications intergation with reviews and requests as service 

### PR DESCRIPTION
**Pull request by serafy0**
### What this PR does (required):
- creates a notification after review
- refactored `createNotification` inside request controller
- the notification service has the parameters 
   - `io` from the controller's `req.io`
   - an object that contains
     - `userId`
     -  `title`
     -  `notificationType` : optional 
     -  `context` : should include the rating or profile picture if exists 

### Screenshots / Videos (required):
Screen Shot 2021-09-20 at 8 47 59 PM

### Any information needed to test this feature (required):
- whenever you post a new review or request, the receiver should get a new notification

### CODE REVIEW

File: server/controllers/review.js [hunk 1]
```diff
@@ -23,6 +25,23 @@ exports.createReview = asyncHandler(async (req, res, next) => {
   const review = await Review.create({ rating, text, profileReviewedId, reviewerId, requestId });
   await Profile.updateOne({ _id: profileReviewedId }, { $push: { reviewsReceived: review } });
 
+  const reviewerProfile = await Profile.findById(reviewerId);
+
+  const requesterFirstNameOrSomeone =
+    reviewerProfile && reviewerProfile.firstName ? reviewerProfile.firstName : 'someone';
```
Comment (ajaytomgeorge) [inline server/controllers/review.js:31]: Sorry, Is this "someone" necessary, is firstName optional?

**Review by ajaytomgeorge — COMMENTED** [review]
Fabulous :-) Just a minor query

File: server/controllers/review.js (hunk 1 above)
Comment (serafy0) [reply server/controllers/review.js:31]: yeah `firstName` is optional. 
I think right now if a new user signs up they would only have an email in their profile.
so It's possible to have firstName as `undefined`

Comment (ajaytomgeorge) [reply server/controllers/review.js:31]: Great, _someone_ is definitely better than _undefined_,

Comment (moffatethan) [reply server/controllers/review.js:31]: I would actually just prefer "Anonymous" than "someone." Bit more common in UI/UX

**Review by moffatethan — APPROVED** [review]

**Pull request merged**